server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -499,10 +499,11 @@ public void validateMapping(Metadata metadata, final IndexMetadata newIndexMetad
             return;
         }
         var tableInfo = tableFactory.create(RelationName.fromIndexName(indexName), metadata);
-        this.getTranslogIndexer = () -> new TranslogIndexer(tableInfo);
+        var indexer = new TranslogIndexer(tableInfo);
+        this.getTranslogIndexer = () -> indexer;
     }
 
-    private TranslogIndexer getTranslogIndexer() {
+    TranslogIndexer getTranslogIndexer() {
```

Comment:
We usually mark the visibility for testing.
```suggestion
    @VisibleForTesting
    TranslogIndexer getTranslogIndexer() {
```